py_gtfs_rt_ingestion/dev_test_setup.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python
+
+import argparse
+import logging
+import sys
+import logging
+from typing import NamedTuple
+import os
+import boto3
+import random
+
+from py_gtfs_rt_ingestion import file_list_from_s3
+
+GTFS_BUCKET = 'mbta-gtfs-s3'
+DEV_INGEST_BUCKET = 'mbta-ctd-dataplatform-dev-incoming'
+DEV_EXPORT_BUCKET = 'mbta-ctd-dataplatform-dev-springboard'
+DEV_ARCHIVE_BUCKET = 'mbta-ctd-dataplatform-dev-archive'
+DEV_ERROR_BUCKET = 'mbta-ctd-dataplatform-dev-error'
+
+logging.basicConfig(level=logging.WARNING)
+
+class SetupArgs(NamedTuple):
+    dev_prefix: str
+    src_prefix: str
+    objs_to_copy: int
+
+def parseArgs(args) -> dict:
+    """
+    parse input args from the command line and generate an event dict in the
+    format the lambda handler is expecting
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--dev-prefix',
+        dest='dev_prefix',
+        type=str,
+        default='lamp/',
+        help='prefix used to save objects in dev environment (lamp) by DEFAULT')
+
+    parser.add_argument(
+        '--source-prefix',
+        dest='src_prefix',
+        type=str,
+        default='',
+        help='prefix of objects to move into dev environment')
+
+    parser.add_argument(
+        '--num',
+        dest='objs_to_copy',
+        type=int,
+        default=10_000,
+        help='number of objects to move into dev environment')
+
+    return SetupArgs(**vars(parser.parse_args(args)))
+
+def clear_dev_buckets(args:SetupArgs):
+    bucket_list = (
+        DEV_INGEST_BUCKET,
+        DEV_EXPORT_BUCKET,
+        DEV_ARCHIVE_BUCKET,
+        DEV_ERROR_BUCKET,
+    )
+    
+    for bucket in bucket_list:
+        uri_root = f"s3://{os.path.join(bucket, args.dev_prefix)}"
+        action = None
+        print(f"Checking for objects to delete in {uri_root}...")
+        files_to_delete = tuple(uri for (uri,size) in file_list_from_s3(bucket, args.dev_prefix) if size > 0)
```

Comment:
why the if condition?

Reply:
and does this capture folders as well?

Reply:
New version does capture folders (except the root lamp folder), so this check is not used anymore.